### PR TITLE
Don't even try to MoveContent to the same window you started in

### PR DESCRIPTION
**Pull request by zadjii-msft**
Don't go.
Tracked in #14957

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -2076,6 +2076,13 @@ namespace winrt::TerminalApp::implementation
         const auto windowId{ args.Window() };
         if (!windowId.empty())
         {
+            // if the windowId is the same as our name, do nothing
+            if (windowId == WindowProperties().WindowName() ||
+                windowId == winrt::to_hstring(WindowProperties().WindowId()))
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TerminalPage.cpp:2081]: Hmmmmm okay, but it does feel a bit weird

**Review by DHowett — APPROVED** [review]

File: src/cascadia/WindowsTerminal/AppHost.cpp [hunk 2]
```diff
@@ -1351,7 +1353,7 @@ void AppHost::_handleMoveContent(const winrt::Windows::Foundation::IInspectable&
         windowBoundsReference = inDips.to_winrt_rect();
     }
 
-    _windowManager.RequestMoveContent(args.Window(), args.Content(), args.TabIndex(), windowBoundsReference);
+    _windowManager.RequestMoveContent(targetWindow, args.Content(), args.TabIndex(), windowBoundsReference);
```
Comment (lhecker) [inline src/cascadia/WindowsTerminal/AppHost.cpp:1356]: This change seems unrelated? The previous code had the benefit that the 3 `args` arguments map more clearly to the function call.

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 1 above)
Comment (lhecker) [reply src/cascadia/TerminalApp/TerminalPage.cpp:2081]: Can you explain why it needs to compare against both properties? I don't quite get that...

Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp:2081]: @lhecker so, the deal is that we generally treat `"window"` as a stealth union. Users can either specify a window name or a window ID in that property. So actions like

```json
        { "keys": "f1", "command": { "action": "moveTab", "window": "1" }, },
        { "keys": "f2", "command": { "action": "moveTab", "window": "foo" }, },
```

both work. 

We don't know whether the string was the ID, or the name, so we have to check both.

File: src/cascadia/WindowsTerminal/AppHost.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/WindowsTerminal/AppHost.cpp:1356]: op yea that was a side effect of the first attempt at a fix

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 1 above)
Comment (DHowett) [reply src/cascadia/TerminalApp/TerminalPage.cpp:2086]: :cry: why can't we just be like, `if(this == otherwindae)`

**Review by lhecker — APPROVED** [review]
